docs/dev-setup.md

```diff
@@ -0,0 +1,92 @@
+# Bisq dev setup guide
+
+This guide describes how to set up a complete Bisq development environment running against a local Bitcoin regtest network. You'll run your own Bisq seed node, arbitration and trading instances in order to allow for end-to-end development and testing.
+
+
+## Prerequisites
+
+Follow the instructions in [build.md](build.md) to build Bisq from source, and it is also recommended to follow the instructions in [idea-import.md](idea-import.md) to be able to run everything from within IntelliJ IDEA. Please also read and follow the instructions in [CONTRIBUTING.md](../CONTRIBUTING.md) prior to submitting any pull requests.
+
+
+## Overview
+
+When developing Bisq, you usually want to use Bitcoin in **regtest** mode and do all networking on **localhost** instead of using the Tor network. Typically, you'll want an environment set up with the following components:
+
+ - Bitcoin Core or bitcoind in regtest mode
+ - A local Bisq seed node
+ - A local Bisq arbitrator instance
+ - 2 local Bisq trading instances (BTC buyer and BTC seller for executing a trade)
+
+You'll set up each of these in the steps that follow.
+
+
+## Run Bitcoin Core (or bitcoind) in regtest mode
+
+The regtest mode operates a local Bitcoin network on your computer. This environment is ideally suited for testing because you are able to create blocks on demand (no need to wait for confirmations) and you don't need to download the blockchain. By creating blocks you act like a miner and you can generate new Bitcoin.
+
+You can find more information about the Bitcoin regtest mode [here](https://bitcoin.org/en/developer-examples#regtest-mode).
+
+Navigate to the [bitcoin.conf](https://en.bitcoin.it/wiki/Running_Bitcoin#Bitcoin.conf_Configuration_File) file and set `regtest=1`, or add `-regtest` as a program argument when starting Bitcoin Core.
+
+At first startup you need to create 101 blocks using the command `generate 101` from the terminal inside Bitcoin Core. 101 blocks are required because of the coin maturity (100 blocks) so you need one more to have at least 50 BTC available for spending.
+
+Later you can create a new blocks with `generate 1`.
+
+
+## Understand Bisq P2P network options
+
+For the local P2P network we prefer to use `localhost`, not the Tor network as it is much faster. But if needed you can combine any of the following combinations of Bitcoin network mode and P2P network mode:
+
+ - localhost + regtest
+ - localhost + testnet
+ - localhost + mainnet
+ - Tor + regtest
+ - Tor + testnet
+ - Tor + mainnet
+
+
+## Understand Bisq program arguments
+
+There are several program arguments required to run in development mode.
```

Comment:
Not required, but a useful argument potentially worth mentioning here is --useDevMode

From https://github.com/bisq-network/bisq/blob/master/common/src/main/java/bisq/common/app/DevEnv.java#L42:

> If set to true we ignore several UI behavior like confirmation popups as well dummy accounts are created and offers are filled with default values. Intended to make dev testing faster.